### PR TITLE
Closes #225 | Add Dataloader ALICE-THI

### LINKED ISSUE #225: Create dataset loader for ALICE-THI
Dataloader name: `alice-thi/alice-thi.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?alice-thi

|     Dataset| alice-thi |
|-------------|---|
| Description | ALICE-THI is a Thai handwritten script dataset that contains 24045 character images, which is split into Thai handwritten character dataset (THI-C68) for 14490 images and Thai handwritten digit dataset (THI-D10) for 9555 images. The data was collected from 150 native writers aged from 20 to 23 years old. The participants were allowed to write only the isolated Thai script on the form and at least 100 samples per character. The character images obtained from this dataset generally have no background noise. |
| Subsets | THI-C68, THI-D10 |
| Languages | tha |
| Tasks | Optical Character Recognition |
| License | Unknown (unknown) |
| Homepage | https://www.ai.rug.nl/~mrolarik/ALICE-THI/ |
| HF URL | - |
| Paper URL | https://www.sciencedirect.com/science/article/abs/pii/S0952197615001724 |

### PR DESCRIPTION
**Pull request by akhdanfadh**
Closes #225 

I implemented one config per language/subset. Thus, configs will look like this: `alice_thi_THI-C68_source`, `alice_thi_THI-D10_seacrowd_imtext`, etc. When testing, pass `alice_thi_<subset>` to the `--subset_id` parameter.

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py`.
- [ ] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

**Comment by yongzx** [comment]
I ran into this issue when I run `python -m tests.test_seacrowd seacrowd/sea_datasets/alice_thi/alice_thi.py --subset alice_thi_THI-D10`. 

```
Traceback (most recent call last):
  File "/Users/yong/Dev/env_seacrowd/lib/python3.8/site-packages/datasets/builder.py", line 1687, in _prepare_split_single
    example = self.info.features.encode_example(record) if self.info.features is not None else record
  File "/Users/yong/Dev/env_seacrowd/lib/python3.8/site-packages/datasets/features/features.py", line 1866, in encode_example
    return encode_nested_example(self, example)
  File "/Users/yong/Dev/env_seacrowd/lib/python3.8/site-packages/datasets/features/features.py", line 1243, in encode_nested_example
    {
  File "/Users/yong/Dev/env_seacrowd/lib/python3.8/site-packages/datasets/features/features.py", line 1244, in <dictcomp>
    k: encode_nested_example(sub_schema, sub_obj, level=level + 1)
  File "/Users/yong/Dev/env_seacrowd/lib/python3.8/site-packages/datasets/features/features.py", line 1243, in encode_nested_example
    {
  File "/Users/yong/Dev/env_seacrowd/lib/python3.8/site-packages/datasets/features/features.py", line 1243, in <dictcomp>
    {
  File "/Users/yong/Dev/env_seacrowd/lib/python3.8/site-packages/datasets/utils/py_utils.py", line 323, in zip_dict
    yield key, tuple(d[key] for d in dicts)
  File "/Users/yong/Dev/env_seacrowd/lib/python3.8/site-packages/datasets/utils/py_utils.py", line 323, in <genexpr>
    yield key, tuple(d[key] for d in dicts)
KeyError: 'context'
```

File: seacrowd/sea_datasets/alice_thi/alice_thi.py [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+                elif self.config.schema == _SEACROWD_SCHEMA:
+                    image_files = list(subfolder.glob("*.png"))
+                    label = int(image_files[0].name.split("-")[0].lower())
+
+                    yield subfolder.name, {
+                        "id": subfolder.name,
+                        "image_paths": [str(file) for file in image_files],
+                        "texts": _SUBSETS[self.config.subset_id]["label_dict"][label],
+                        "metadata": {
+                            # "context": "",
```
Comment (yongzx) [inline seacrowd/sea_datasets/alice_thi/alice_thi.py:258]: I suppose this is the reason that caused the error in unit tests.

**Comment by ljvmiranda921** [comment]
Will review this after addressing @yongzx 's comments! Got a bit busy this week 😅

**Comment by akhdanfadh** [comment]
I don't know why the comment resulted in an error on your end, but not on mine. I've uncommented the line there. Also, ran the makefile.

@yongzx @ljvmiranda921

**Review by ljvmiranda921 — APPROVED** [review]
Ran the source subset_ids and it passed! lgtm

**Comment by yongzx** [comment]
It runs for me now! LGTM

**Pull request merged**